include/envoy/filesystem/filesystem.h

```diff
@@ -102,6 +102,12 @@ class Instance {
    */
   virtual std::string fileReadToEnd(const std::string& path) PURE;
 
+  /**
+   * @path full file path on input which is truncated to the path
+   * @name the resulting file name component from the input path
+   */
+  virtual void splitFileName(std::string& path, std::string& name) PURE;
```

Comment:
From an interface perspective would it be better to take a string_view of the thing to split, and then return 2 string views in a struct that provide the split info? This seems a bit more flexible to allow a constant input?

Reply:
From a storage allocation perspective, splitFileName is often recursively called, e.g. creating the path /foo/bar/bash, when bash doesn't exist, the function would then split and evaluate /foo/bar, and if that doesn't exist, split and evaluate /foo.

Also, splitFileName used in any sort of file list/tree context needs only one allocated name storage of size 255 characters, to avoid one string-per-item allocation. (If not preallocated, it can conceivably re-allocate on each call.)

The proposed interface avoids all reallocation of the path storage, and avoids allocation of the name storage if it fits any preallocation of the name variable.

We are open to returning two strings, as long as the memory growth traversing long paths or iterating on a many member file list is recognized.

Reply:
I'm not following, can't you just return views within the initial string? Why are any allocations required at all?

Reply:
Understood! Thanks for the suggestion, we'll address this.